tests/helpers/kube.go

```diff
@@ -580,3 +583,109 @@ func GetPodsOnNode(ctx context.Context, client *resources.Resources, nodeName st
 
 	return podList.Items, nil
 }
+
+// WaitForNodeConditionWithCheckName waits for the node to have a condition with the reason containing the specified checkName.
+func WaitForNodeConditionWithCheckName(ctx context.Context, t *testing.T, c klient.Client, nodeName, checkName string) {
+	require.Eventually(t, func() bool {
+		node, err := GetNodeByName(ctx, c, nodeName)
+		if err != nil {
+			t.Logf("failed to get node %s: %v", nodeName, err)
+			return false
+		}
+
+		// Look for a condition where the reason contains the check name
+		for _, condition := range node.Status.Conditions {
+			if condition.Status == v1.ConditionTrue && strings.Contains(condition.Reason, checkName) {
+				t.Logf("Found node condition: Type=%s, Reason=%s, Status=%s, Message=%s",
+					condition.Type, condition.Reason, condition.Status, condition.Message)
+				return true
+			}
+		}
+
+		t.Logf("Node %s does not have a condition with check name '%s'", nodeName, checkName)
+		return false
+	}, WaitTimeout, WaitInterval, "node %s should have a condition with check name %s", nodeName, checkName)
+}
+
+// CleanupNodeConditionAndUncordon clears node conditions with the specified checkName by setting them to healthy state and uncordons the node.
+// NOTE: this function was added specifically to clean up node conditions added by health event analyzer.
+func CleanupNodeConditionAndUncordon(ctx context.Context, c klient.Client, nodeName, checkName string) error {
+	clientset, err := kubernetes.NewForConfig(c.RESTConfig())
+	if err != nil {
+		return fmt.Errorf("failed to create clientset: %w", err)
+	}
+
+	maxRetries := 3
+	var lastErr error
+
+	for attempt := 1; attempt <= maxRetries; attempt++ {
+		node, err := clientset.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get node %s: %w", nodeName, err)
+		}
+
+		conditionFound := false
+		conditionUpdated := false
+
+		for i, condition := range node.Status.Conditions {
+			if strings.Contains(string(condition.Type), checkName) || strings.Contains(condition.Reason, checkName) {
+				klog.Infof("Found condition to clean up: Type=%s, Reason=%s, Status=%s, Message=%s",
+					condition.Type, condition.Reason, condition.Status, condition.Message)
+				conditionFound = true
+				if condition.Status == v1.ConditionTrue {
+					node.Status.Conditions[i].Status = v1.ConditionFalse
+					node.Status.Conditions[i].Message = "No Health Failures"
+					node.Status.Conditions[i].Reason = fmt.Sprintf("%sIsHealthy", checkName)
+					node.Status.Conditions[i].LastHeartbeatTime = metav1.Now()
+					node.Status.Conditions[i].LastTransitionTime = metav1.Now()
+					conditionUpdated = true
+				}
+			}
+		}
+
+		if !conditionFound {
+			// No condition to clean up
+			break
+		}
+
+		if conditionUpdated {
+			_, err = clientset.CoreV1().Nodes().UpdateStatus(ctx, node, metav1.UpdateOptions{})
```

Comment:
Not a fan of doing this, can we instead just send a health event and make sure it workes or fail if it doesn't?